src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MetricsCollectorToolSpark.java

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.metrics.MetricsArgs;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Base class for standalone Spark metrics collector tools.ß
+ *
+ */
+public abstract class MetricsCollectorToolSpark<T extends MetricsArgs>
```

Comment:
I also think it's a bit strange that this implements MetricsCollectorSpark.  Could it be refactored to just contain a `MetricsCollectorSpark` instead of being one?  Is there a use case where you would want to use the `MetricsCollectorToolSpark` directly as a metrics collector?